client/components/editor/teaching-elements/Assessment/TextSelection/Selections.js

```diff
@@ -0,0 +1,62 @@
+import { extractIndex } from './helpers';
+import cloneDeep from 'lodash/cloneDeep';
+import curry from 'lodash/curry';
+import curryRight from 'lodash/curryRight';
+import first from 'lodash/first';
+import inRange from 'lodash/inRange';
+import last from 'lodash/last';
+
+export default class Selections {
+  constructor(selection = []) {
+    this.ranges = cloneDeep(selection);
+  }
+
+  merge(selection) {
+    const begin = first(selection);
+    const final = last(selection);
+    const check = curry(isOverlaped);
+    const beginOverlap = this.ranges.find(check(begin));
+    const endOverlap = this.ranges.find(check(final - 1));
+    const range = [first(beginOverlap) || begin, last(endOverlap) || final];
+    this.remove(range);
+    this.ranges.push(range);
+    return range;
+  }
+
+  remove(range) {
+    const check = curryRight(isOverlaped)(range);
+    this.ranges = this.ranges.filter(range => (
+      !check(first(range)) &&
+      !check(last(range) - 1)
+    ));
+  }
+
+  getByIndex(index) {
+    const check = curry(isOverlaped)(index);
+    return this.ranges.find(check);
+  }
+
+  processNode(node) {
+    const index = extractIndex(node);
+    const selection = this.getByIndex(index);
+    if (!selection) return node;
+    node.classList.add('selected');
+    node.setAttribute('range', selection);
+    return node;
+  }
+
+  static extractRange(node) {
+    const range = node.getAttribute('range');
+    return range.split(',').map(Number);
+  }
+
+  static isSelected(node) {
+    return node.hasAttribute('range');
+  }
+}
```

Comment:
You are actually violating Single Responsibility Principle here 🙃 
1. `Selections` class works with virtual data structure, while
2. `processNode`, `extractRange` and `isSelected` touch specific (word) DOM nodes (plus imported `extractRange` helper)

Divide that by responsibilities :bowtie: